inst/stan/epinowcast.stan

```diff
@@ -89,26 +90,30 @@ transformed parameters{
   real phi; // Transformed overdispersion (joint across all observations)
   // calculate log mean and sd parameters for each dataset from design matrices
   profile("transformed_delay_reference_date_total") {
-  profile("transformed_delay_reference_date_effects") {
-  logmean = combine_effects(logmean_int, logmean_eff, d_fixed, logmean_sd,
-                            d_random);
-  logsd = combine_effects(log(logsd_int), logsd_eff, d_fixed, logsd_sd,
-                          d_random); 
-  logsd = exp(logsd);
-  }
-  // calculate pmfs
-  profile("transformed_delay_reference_date_pmfs") {
-  for (i in 1:npmfs) {
-    pmfs[, i] = discretised_reporting_prob(logmean[i], logsd[i], dmax, dist);
-  }
-  if (ref_p == 0) {
+  if (dist) {
+    profile("transformed_delay_reference_date_effects") {
+    logmean = combine_effects(logmean_int[1], logmean_eff, d_fixed, logmean_sd,
+                              d_random);
+    if (dist > 1) {
+      logsd = combine_effects(log(logsd_int[1]), logsd_eff, d_fixed, logsd_sd,
+                              d_random); 
+      logsd = exp(logsd);
+    }
+    }
+    // calculate pmfs
+    profile("transformed_delay_reference_date_pmfs") {
     for (i in 1:npmfs) {
-      ref_lh[, i] = prob_to_hazard(pmfs[, i]);
-      ref_lh[, i] = logit(ref_lh[, i]);
+      pmfs[, i] = discretised_reporting_prob(logmean[i], logsd[i], dmax, dist);
+    }
+    if (ref_p == 0) {
+      for (i in 1:npmfs) {
+        ref_lh[, i] = prob_to_hazard(pmfs[, i]);
+        ref_lh[, i] = logit(ref_lh[, i]);
+      }
+    }else{
+      ref_lh = pmfs;
+    }
     }
-  }else{
-    ref_lh = pmfs;
-  }
   }
```

Comment:
I'm a bit confused: if dist==0 (exponential pmf), the reporting probs and ref_lh are never computed, are they? Also, why would we not want to support a reference day model for exponential delays? The hazard is only constant over the delays for a given reference date, but it should be able to change over the reference dates. Based on Sebastian's proposal I would have expected to have dist==0 and dist==exponential to be two different options.